packages/dd-trace/src/profiling/profiler.js

```diff
@@ -20,13 +20,27 @@ class Profiler extends EventEmitter {
     this._timer = undefined
     this._lastStart = undefined
     this._timeoutInterval = undefined
+    this._resetStarted()
+  }
+
+  started () {
+    return this._started
   }
 
   start (options) {
-    this._start(options).catch((err) => { if (options.logger) options.logger.error(err) })
+    const started = this._start(options)
+    started.catch((err) => { if (options.logger) options.logger.error(err) })
+    started.then(this._startedResolve, this._startedReject)
```

Comment:
Why not make start directly return the promise from the async function ?
```suggestion
    return this._start(options).catch((err) => { if (options.logger) options.logger.error(err) })
```

This way no need for `this._started`.

Reply:
hm… fair enough. In that case anyone calling `Proxy.profilerStarted()` would get `undefined` before `Proxy.init()` but that might be tolerable.

Reply:
Hm… you're right that it might make sense to have a promise that returns a boolean, specifically because the tracing can be initialized without profiling. However, I don't like returning `Promise.resolve(false)` before the profiler is started; that feels rather useless. If the profiler is started afterwards, it becomes a different promise, so one can't use the first promise to wait for an eventual profiler start fulfillment… in this case, we might as well have a function that just returns a boolean and have the caller poll it periodically… 

I'm also fine with `profilerStarted` only being valid after `init` was called, even throw an error when it's used before it.